extensions/binding/org.eclipse.smarthome.binding.ntp/src/main/java/org/eclipse/smarthome/binding/ntp/handler/NtpHandler.java

```diff
@@ -0,0 +1,196 @@
+
+	/** NTP host */
+	private String hostname;
+	/** refresh interval */
+	private BigDecimal refreshInterval;
+	/** NTP refresh frequency */
+	private BigDecimal refreshNtp = new BigDecimal(0);
+	/** Timezone */
+	private TimeZone timeZone;
+	/** Locale */
+	private Locale locale;
+
+	/** NTP refresh counter */
+	private int refreshNtpCount = 0;
+	/** NTP system time delta */
+	private long timeOffset;
+
+	public NtpHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		// No specific commands tied to this, but we will trigger an update
+		this.refreshNtpCount = 0;
+		refreshTimeDate();
+	}
+
+	@Override
+	public void initialize() {
+
+		try {
+			logger.debug("Initializing NTP handler for '{}'.",getThing().getUID());
+
+			Configuration config = getThing().getConfiguration();
+			this.hostname = (String) config.get(PROPERTY_NTP_SERVER);
+			this.refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
+			this.refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
+			this.refreshNtpCount = 0;
+
+			try {
+				timeZone = TimeZone.getTimeZone((String) config.get(PROPERTY_TIMEZONE));
+			} catch (Exception e) {
+				timeZone = TimeZone.getDefault();
+				logger.debug("using default TZ: {}", timeZone);
+			}
+
+			try {
+				String localeString = (String) config.get(PROPERTY_LOCALE);
+				locale = new Locale(localeString);
+			} catch (Exception e) {
+				locale = Locale.getDefault();
+				logger.debug("using default locale: {}", locale);
+			}
+
+			updateStatus(ThingStatus.ONLINE);
+			startAutomaticRefresh();
+
+		} catch (Exception ex) {
+			logger.error("Error occured while initializing NTP handler: " + ex.getMessage(), ex);
+			updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+		}
+	}
+
+	@Override
+	public void dispose() {
+		refreshJob.cancel(true);
+		super.dispose();
+	}
+
+	private void startAutomaticRefresh() {
+
+		Runnable runnable = new Runnable() {
+			@Override
+			public void run() {
+				try {
+					refreshTimeDate();
+				} catch (Exception e) {
+					logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+				}
+			}
+		};
+
+		refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, refreshInterval.intValue(), TimeUnit.SECONDS);
+	}
+
+	private synchronized void refreshTimeDate() {
+
+		long networkTimeInMillis;
+		if (refreshNtpCount <= 0) {
+			networkTimeInMillis = getTime(hostname);
+			timeOffset = networkTimeInMillis - System.currentTimeMillis();
+			logger.debug("delta system time: {}", timeOffset);
+			refreshNtpCount = refreshNtp.intValue();
+		} else {
+			networkTimeInMillis = System.currentTimeMillis() + timeOffset;
+			refreshNtpCount--;
+		}
+
+		Calendar calendar = Calendar.getInstance(timeZone, locale);
+		calendar.setTimeInMillis(networkTimeInMillis);
+
+		updateState(new ChannelUID(getThing().getUID(), CHANNEL_DATE_TIME), new DateTimeType(calendar));

```

Comment:
probably better to define the channelUID instance as a field and reuse it here to avoid to many class instantiations.